rb_ws/src/buggy/scripts/2d_sim/engine.py

```diff
@@ -87,15 +94,20 @@ def update_steering_angle(self, data: Float64):
         with self.lock:
             self.steering_angle = data.data
 
-    def update_velocity(self, data: Float64):
+    def update_velocity(self, data: Float64, source):
         """Updates the velocity as callback function for subscriber
 
         Args:
             data (Float64): velocity in m/s
+            source (string): whether incoming data is manual or simulated
         """
-        with self.lock:
-            self.velocity = data.data
-
+        if (source == "simulated" and not Simulator.MANUAL_V_FLAG):
+            with self.lock:
+                self.velocity = data.data
+        if (source == "manual" and Simulator.MANUAL_V_FLAG):
+            with self.lock:
+                self.velocity = data.data
```

Comment:
else case?

Reply:
I think the intention here is that if the velocity is only updated if the source topic matches the mode flag. So if neither cases match, velocity should not receive an update.

Reply:
update_velocity() is only called when velocity needs to be updated right (as a callback function). So I think an else case here would catch cases where there's clearly an error. Just a safety thing, perhaps a ROS_WARN gets raised?